app/scripts/metamask-controller.js

```diff
@@ -219,6 +221,31 @@ export default class MetamaskController extends EventEmitter {
       state: initState.CurrencyController,
     });
 
+    const tokenListMessenger = controllerMessenger.getRestricted({
+      name: 'TokenListController',
+    });
+    this.tokenListController = new TokenListController({
+      chainId: hexToDecimal(this.networkController.getCurrentChainId()),
+      useStaticTokenList: this.preferencesController.store.getState()
+        .useStaticTokenList,
+      onNetworkStateChange: (cb) =>
+        this.networkController.store.subscribe(async (networkState) => {
+          const modifiedNetworkState = {
+            ...networkState,
+            provider: {
+              ...networkState.provider,
+              chainId: hexToDecimal(networkState.provider.chainId),
+            },
+          };
+          return await cb(modifiedNetworkState);
```

Comment:
[REQ] if cb is a promise returning function then I think the async/await here isn't necessary right? If you return `cb(modifiedNetworkState)` it'll return the promise from the `cb` method which can be awaited in the controller. I could be missing something though!